ui/docs/manual/using/integration/frontend.md

```diff
@@ -88,112 +66,108 @@ const status = document.getElementById('status');
-  clientID: '<YOUR AUTH0 CLIENT ID>',
-  audience: '<YOUR AUTH0 CLIENT AUDIENCE>',
-  // This redirect URI is configured with your auth0 client.
-  // For this example, we will pretend that our current HTML
-  // page is going to be served no matter what route is used
-  redirectUri: '<YOUR AUTH0 CLIENT REDIRECT URL>'
-});
-```
+| Parameter | Required | Description |
+| --- | --- | --- |
+| response_type |✓ |The value MUST be "code" for requesting an authorization code, or "token” for requesting an access token. |
+| client_id | ✓ | The registered OAuth client. |
+| redirect_uri | ✓ | The URL to which the browser should be redirected after authorization. |
+| scope | ✓ | | An array of URIs to which the server is allowed to redirect the user. |
+| state | Optional (strongly recommended) | An opaque value used by the client to maintain state between the request and callback.  The authorization server includes this value when redirecting the user-agent back to the client.  The parameter SHOULD be used for preventing cross-site request forgery. |
+| expires | Optional | The requested lifetime of the resulting Taskcluster credentials, in a format understood by [`fromNow`](), defaulting to the registered OAuth client `maxExpires` value. |
+
+
+Once the sign in is complete, Taskcluster will
+redirect back to the `redirect_uri` you specified in the URL.
+The redirect URI will have the authorization `code` in the
+query parameter which will then be used to grab a third party access token.
+In the case of an implicit grant (i.e., `response_type=“token”`) the query
+parameter will instead have the access token so you can safely skip the next
+section and continue at [Exchanging a Third Party token for
+Taskcluster Credentials](#exchanging-a-third-party-token-for-taskcluster-credentials) section.
+
+### Exchanging a code for a third-party access token
 
-With our auth0 client instance we can now respond to actions. Let's capture
-clicks on our action link:
+Next, the client requests an access token from the authorization server’s token endpoint by
+including the authorization code received in the previous step.
+The token endpoint is `<rootUrl>/login/oauth/token`.
 
 ```js
-// We are going to get an access token from auth0
-// which we can give to Taskcluster. Let's store
-// that here when we get it
-let accessToken;
+if (window.location.search) {
+  const qs = parseQuery(window.location.search);
 
-link.addEventListener('click', (e) => {
-  // make sure the page doesn't redirect on click
-  e.preventDefault();
-  
-  // If we don't yet have out access token, let's go
-  // get it from auth0 by having the user sign in
-  if (!accessToken) {
-    return auth.authorize();
+  if (qs.error) {
+    // The authorization server responded with an error.
+    // The list of error codes can be found in https://tools.ietf.org/html/rfc6749#section-4.1.2.1
+    status.innerText = qs.error;
+  } else if (qs.state === '99') {
+    fetchToken(qs.code).then(data => {
```

Comment:
should this function definition be included here?  Or maybe just a comment saying "see below"?